packages/camera/tizen/src/camera_device.h

```diff
@@ -159,23 +224,45 @@ class CameraDevice {
   bool SetCameraAutoFocusMode(CameraAutoFocusMode mode);
   bool SetCameraAutoFocusArea(int x, int y);
   bool SetCameraAutoFocusChangedCb(CameraFocusChangedCb callback);
-  bool SetCameraMediaPacketPreviewCb(MediaPacketPreviewCb callback);
+  bool SetCameraMediaPacketPreviewCb(CameraMediaPacketPreviewCb callback);
   bool SetCameraPreviewCb(CameraPrivewCb callback);
   bool SetCameraPreviewFormat(CameraPixelFormat format);
   bool SetCameraPreviewSize(Size size);
   bool SetCameraZoom(int zoom);
-  bool StartCameraCapture(OnCaptureSuccessCb on_success,
-                          OnCaptureFailureCb on_failure);
+  bool StartCameraCapture(const OnCaptureSuccessCb &on_success,
+                          const OnCaptureFailureCb &on_failure);
   bool StartCameraAutoFocusing(bool continuous);
   bool StartCameraPreview();
   bool StopCameraAutoFocusing();
   bool StopCameraPreview();
   bool UnsetCameraMediaPacketPreviewCb();
   bool UnsetCameraAutoFocusChangedCb();
 
-  bool PrintSupportedPreviewResolution();
-  void PrintState();
-  bool PrintPreviewRotation();
+  bool CancleRecorder();
+  bool CreateRecorder();
+  bool CommitRecorder();
+  bool DestroyRecorder();
+  bool GetRecorderState(RecorderState &state);
+  bool GetRecorderFileName(std::string &name);
+  bool SetRecorderAudioChannel(RecorderAudioChannel chennel);
+  bool SetRecorderAudioDevice(RecorderAudioDevice device);
+  bool SetRecorderAudioEncorder(RecorderAudioCodec codec);
+  bool SetRecorderAudioSamplerate(int samplerate);
+  bool SetRecorderFileFormat(RecorderFileFormat format);
+  bool SetRecorderFileName(std::string name);
```

Comment:
How about changing std::string to const std::string& ?

Reply:
Oh! thank you